src/components/feedback-indicators/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from 'react'
+import _ from 'lodash'
+import { Animated, View } from 'react-native'
+import { useStyles, useTheme } from '../../../theme'
+import { shameStyles } from '../../../theme/shame-styles'
+
+export interface ProgressBarProps {
+  /**
+   * Progression of the task. Number value between [0, 100].
+   */
+  progress: number
+}
+
+const { height } = shameStyles.progressBar
+
+/**
+ * Use this component to visually represent the completion of a task or operation.
+ */
+export const ProgressBar: React.FC<ProgressBarProps> = ({ progress: progressRaw }) => {
+  const styles = useStyles(theme => ({
+    container: {
+      backgroundColor: theme.colors.fill.background.darker,
+      borderRadius: theme.radius.medium,
+      overflow: 'hidden',
+      width: '100%',
+      height,
+    },
+    progress: {
+      position: 'absolute',
+      left: '-100%', // Make progress bar twice the width to be able to animate width using scaleX
```

Comment:
Using positional values and scaling seems a bit hackish to me.

Tried going with `width` instead. Here's what I ended up with:

```tsx
export const ProgressBar: React.FC<ProgressBarProps> = ({ progress }) => {
  const styles = useStyles(theme => ({
    container: {
      backgroundColor: theme.colors.fill.background.darker,
      borderRadius: theme.radius.medium,
      overflow: 'hidden',
      width: '100%',
      height,
    },
    progress: {
      position: 'absolute',
      width: '100%', // <-- starts at full width
      left: 0,
      right: 0,
      top: 0,
      bottom: 0,
      backgroundColor: theme.colors.fill.accent.default,
    },
  }))

  const anim = useRef(new Animated.Value(0)) // <-- ref to animate from current position
  const { animation } = useTheme()

  useEffect(() => {
    Animated.timing(anim.current, {
      easing: animation.easing.move,
      toValue: progress,
      duration: animation.duration.shorter,
      useNativeDriver: true,
    }).start()
  }, [progress])

  const width = anim.current.interpolate({
    inputRange: [0, 100],
    outputRange: ['0%', '100%'], // <-- interpolate between % values
    extrapolate: 'clamp', // <-- clamp during interpolation (to avoid "progressRaw")
  })

  return (
    <View style={styles.container}>
      <Animated.View style={[styles.progress, { width }]} />
    </View>
  )
}
```

Reply:
Animating width using `useNativeDriver: true` won't work on Native apps (iOS & Android). This is why I kept the absolute positioning + scaling animation.